mlx/fast.cpp

```diff
@@ -859,4 +864,229 @@ array affine_dequantize(
   return fallback({w, scales, biases})[0];
 }
 
+void MetalKernel::validate_output_shapes() {
+  // Make sure output shapes and dtypes have the same keys
+  bool validated = true;
+  if (output_shapes_.size() == 0) {
+    throw std::invalid_argument("Must specify at least one output.");
+  }
+  if (output_shapes_.size() != output_dtypes_.size()) {
+    validated = false;
+  } else {
+    for (auto kv : output_shapes_) {
+      if (output_dtypes_.find(kv.first) == output_dtypes_.end()) {
+        validated = false;
+      }
+    }
+  }
+  if (!validated) {
+    throw std::invalid_argument(
+        "`output_shapes` and `output_dtypes` must have the same keys.");
```

Comment:
Could you add the `[...]` as a prefix to the error message for consistency with other error messages. Not sure what should go inside yet something like `[metal_kernel] `output_shapes` ...`